src/libstd/sync/mpsc/shared.rs

```diff
@@ -216,7 +216,7 @@ impl<T> Packet<T> {
         Ok(())
     }
 
-    pub fn recv(&mut self) -> Result<T, Failure> {
+    pub fn recv_impl(&mut self, deadline: Option<Instant>) -> Result<T, Failure> {
```

Comment:
Perhaps this could just be the only exported function of each of these modules? e.g.:

```rust
pub fn recv(&mut self, deadline: Option<Instant>) -> ...
```